src/java.base/share/classes/sun/security/util/Debug.java

```diff
@@ -208,16 +269,53 @@ public void println(Object obj, String message)
 
     public void println()
     {
-        System.err.println(prefix + ":");
+        System.err.println(prefix + extraInfo() + ":");
     }
 
     /**
      * print a message to stderr that is prefixed with the prefix.
      */
 
-    public static void println(String prefix, String message)
+    public void println(String prefix, String message)
     {
-        System.err.println(prefix + ": "+message);
+        System.err.println(prefix + extraInfo() + ": " + message);
+    }
+
+    /**
+     * If thread debug option enabled, include information containing
+     * hex value of threadId and the current thread name
+     * If timestamp debug option enabled, include timestamp string
+     * @return extra info if debug option enabled.
+     */
+    private String extraInfo() {
+        String retString = "";
+        if (printThreadDetails) {
+            retString = toHexString(Thread.currentThread().threadId()) + "|" +
+                    Thread.currentThread().getName() + "|" + formatCaller();
+        }
+        if (printDateTime) {
+            retString += (retString.isEmpty() ? "" : "|")
+                    + FormatHolder.DATE_TIME_FORMATTER.format(Instant.now());
+        }
+        return retString.isEmpty() ? "" : "[" + retString + "]";
+
+    }
+
+    // copied from sun/security/ssl/Utilities.java for now
```

Comment:
Why own helper, string.format(%x) can build the whole string or call Long.toHexString() or toString(,16)?